### PR TITLE
RegisterMetadata: support compute pipeline and task shader pipeline

### PR DESCRIPTION
**Pull request by xuechen417**
Implement `buildCsRegister()` and modify `buildShaderExecutionRegisters` to support `HardwareStages::Cs`.

### CODE REVIEW

**Comment by xuechen417** [comment]
retest this please

**Comment by xuechen417** [comment]
V1: rebase

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6c51406b6c92f18cd6b0a810b71f8ed8ae1855e1
<details><summary>CTS tests (Failed: 0/211523)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39249/70507 (55.7%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31258/70507 (44.3%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39499/70507 (56.0%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31008/70507 (44.0%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36346/70509 (51.5%)</li>
<li>Failed: 0/70509 (0.0%)</li>
<li>Not Supported: 34163/70509 (48.5%)</li><li>Warnings: 0/70509 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/ConfigBuilderBase.h [hunk 1]
```diff
@@ -129,6 +130,9 @@ class ConfigBuilderBase {
   llvm::msgpack::MapDocNode m_graphicsRegistersNode;
   // MsgPack map node for graphics registers metadata
   // ".graphics_registers"
+  llvm::msgpack::MapDocNode m_computeRegistersNode;
+  // MsgPack map node for graphics registers metadata
+  // ".compute_registers"
```
Comment (nhaehnle) [inline lgc/patch/ConfigBuilderBase.h:135]: If comments are not on the same line, can you please put them *before* (and keep the line length in mind)? That is:
```suggestion
  // MsgPack map node for graphics registers metadata ".compute_registers"
  llvm::msgpack::MapDocNode m_computeRegistersNode;
```
I realize that the pattern already exists above -- I hadn't noticed it before, sorry.

File: lgc/patch/RegisterMetadataBuilder.cpp [hunk 2]
```diff
@@ -856,8 +864,45 @@ void RegisterMetadataBuilder::buildPsRegisters() {
...
 // Builds register configuration for compute/task shader.
-void RegisterMetadataBuilder::buildCsRegisters() {
-  // TODO: Implement once PAL completes the new format for CS
+void RegisterMetadataBuilder::buildCsRegisters(ShaderStage shaderStage) {
+  assert(shaderStage == ShaderStageCompute || shaderStage == ShaderStageTask);
+
+  getComputeRegNode()[Util::Abi::ComputeRegisterMetadataKey::TgidXEn] = true;
+  getComputeRegNode()[Util::Abi::ComputeRegisterMetadataKey::TgidYEn] = true;
+  getComputeRegNode()[Util::Abi::ComputeRegisterMetadataKey::TgidZEn] = true;
+  getComputeRegNode()[Util::Abi::ComputeRegisterMetadataKey::TgSizeEn] = true;
```
Comment (nhaehnle) [inline lgc/patch/RegisterMetadataBuilder.cpp:873]: Do you plan to optimize this later?

**Review by nhaehnle — COMMENTED** [review]
Looks pretty good, one nit inline.

File: lgc/patch/RegisterMetadataBuilder.cpp (hunk 2 above)
Comment (xuechen417) [reply lgc/patch/RegisterMetadataBuilder.cpp:873]: Do you mean we should set `TgidZEn` for compute shaders based on `foldWorkgroupXY`?

**Review by nhaehnle — APPROVED** [review]

**Pull request merged**